Application Provisioning Connection Profile update: add OAUTH2 support for authScheme (#3012)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/apps/index.md

| ---------------- | ------------------------------------------------------------ | --------------------------------------------------------------------------- | -------- | ------ | -------- | --------- |
| authScheme              | Defines the method of authentication    | `TOKEN`, `OAUTH2` <ApiLifecycle access="ea" /> , `UNKNOWN`                                                           | FALSE    | FALSE  | TRUE    |           |
| _links            | Discoverable resources related to the connection            | [JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)              | TRUE    | FALSE   | TRUE    |           |
| status            | Status of the connection      | `ENABLED`, `DISABLED`, `UNKNOWN`  | FALSE    | FALSE   | TRUE    | `DISABLED` |

If the authScheme is `UNKNOWN`, then either the authentication scheme used by the application isn't supported or the the application doesn't support provisioning. An object with an `UNKNOWN` `authScheme` results in an `UNKNOWN` `status`.

### Provisioning Connection Profile object

> The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, both token based and  OAuth 2.0 based authentication <ApiLifecycle access="ea" /> are supported.

Reviewer: ```suggestion
The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, both token-based and OAuth 2.0-based authentication <ApiLifecycle access="ea" /> are supported.
```

> #### Token based Provisioning Connection Profile example

Reviewer: ```suggestion
#### Token-based Provisioning Connection Profile example
```

| ---------------- | ------------------------------------------------------------ | --------------------------------------------------------------------------- | -------- | ------ | -------- | --------- |
| authScheme              | Defines the method of authentication    | `TOKEN`, `OAUTH2` <ApiLifecycle access="ea" /> , `UNKNOWN`                                                           | FALSE    | FALSE  | TRUE    |           |
| _links            | Discoverable resources related to the connection            | [JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)              | TRUE    | FALSE   | TRUE    |           |
| status            | Status of the connection      | `ENABLED`, `DISABLED`, `UNKNOWN`  | FALSE    | FALSE   | TRUE    | `DISABLED` |

If the authScheme is `UNKNOWN`, then either the authentication scheme used by the application isn't supported or the the application doesn't support provisioning. An object with an `UNKNOWN` `authScheme` results in an `UNKNOWN` `status`.

### Provisioning Connection Profile object

> The application provisioning connection profile is used to configure the method of authentication and the credentials. Currently, both token-based and OAuth 2.0-based authentication <ApiLifecycle access="ea" /> are supported.

Reviewer 2: You might change:
"Currently, both token-based and OAuth...."
to
"Currently, token-based and OAuth..."

Just thinking we don't need to specify 'both' as we list which options are currently available in the description.

| authScheme              | Defines the method of authentication     | `TOKEN` | FALSE    | FALSE  | FALSE    |
| token            | Token used to authenticate with application      | String | FALSE    | FALSE   | FALSE    |

> #### OAuth 2.0 based Provisioning Connection Profile example

Reviewer: ```suggestion
#### OAuth 2.0-based Provisioning Connection Profile example
```

```json
{
    "profile": {
        "authScheme": "OAUTH2",
        "clientId": "${clientId}"
    }
}
```

> #### OAuth 2.0 based Provisioning Connection Profile properties

Reviewer: ```suggestion
#### OAuth 2.0-based Provisioning Connection Profile properties
```
Maybe add `<ApiLifecycle access="ea" />` below this header to apply EA flag to entire table?

| Property                                | Description                                                   | DataType                               | Nullable | Unique | Readonly |
| --------------------------------------- | ------------------------------------------------------------- | -------------------------------------- | -------- | ------ | -------- | --------- | --------- | ---------- |
| authScheme                              | Defines the method of authentication                          | `OAUTH2` <ApiLifecycle access="ea" />  | FALSE    | FALSE  | FALSE    |

> | clientId <ApiLifecycle access="ea" />   | Unique identifier for OAuth 2.0 service app in the target org | String                                 | FALSE    | FALSE  | FALSE    |

Reviewer: ```suggestion
| clientId <ApiLifecycle access="ea" />   | Unique client identifier for the OAuth 2.0 service app from the target org | String                                 | FALSE    | FALSE  | FALSE    |
```
